From af48ae3cf2f069be9bed756ccf0fbcc28c28a3b5 Mon Sep 17 00:00:00 2001
From: Iain Lane <iain@orangesquash.org.uk>
Date: Fri, 23 Jan 2015 17:05:40 +0000
Subject: [PATCH] gtk-reftest: Force icon theme to Adwaita

Since reftests are developed against Adwaita, they sometimes fail on
other icon themes.
---
 testsuite/reftests/gtk-reftest.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/testsuite/reftests/gtk-reftest.c b/testsuite/reftests/gtk-reftest.c
index 19f3899..d3e106a 100644
--- a/testsuite/reftests/gtk-reftest.c
+++ b/testsuite/reftests/gtk-reftest.c
@@ -333,6 +333,7 @@ int
 main (int argc, char **argv)
 {
   const char *basedir;
+  GtkSettings *settings;
   
   /* I don't want to fight fuzzy scaling algorithms in GPUs,
    * so unless you explicitly set it to something else, we
@@ -343,6 +344,12 @@ main (int argc, char **argv)
   if (!parse_command_line (&argc, &argv))
     return 1;
 
+  settings = gtk_settings_get_default ();
+
+  g_assert_nonnull (settings);
+
+  gtk_settings_set_string_property (settings, "gtk-icon-theme-name", "Adwaita", "");
+
   if (arg_base_dir)
     basedir = arg_base_dir;
   else
-- 
2.1.4

